### PR TITLE
Support get entity by ids in segcore

### PR DESCRIPTION
**Pull request by FluorineDog**
Support GetEntityByIds in Segcore

Signed-off-by: fluorinedog <fluorinedog@gmail.com>

### CODE REVIEW

File: internal/core/src/segcore/ScalarIndex.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        dst_ids->add_data(entry_id);
+        dst_offsets.push_back(entry_offset);
+    }
+    return {std::move(res_ids), std::move(dst_offsets)};
+}
+void
+ScalarIndexVector::append_data(const ScalarIndexVector::T* ids, int64_t count, SegOffset base) {
+    for (int64_t i = 0; i < count; ++i) {
+        auto offset = base + SegOffset(i);
+        mapping_.emplace_back(ids[i], offset);
```
Comment (yhmo) [inline internal/core/src/segcore/ScalarIndex.cpp:54]: Is there possible multi-threads contest on the "mapping_" ?

**Review by yhmo — APPROVED** [review]

**Review by shengjun1985 — APPROVED** [review]

**Review by wxyucs — APPROVED** [review]

File: internal/core/src/segcore/ScalarIndex.cpp (hunk 1 above)
Comment (FluorineDog) [reply internal/core/src/segcore/ScalarIndex.cpp:54]: No data race possible, because append_data will be used in single thread

**Pull request merged**